spec/controllers/hq/sponsorships_controller_spec.rb

```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+RSpec.describe Hq::SponsorshipsController, type: :controller do
+  let(:sponsorship) {create :sponsorship, active: true}
+
+  before :each do
+    sign_in instance_double(AdminUser)
+  end
+
+  specify "#inactivate" do
+    expect(Sponsorship).to receive(:find).and_return(sponsorship)
+    put :inactivate, id: sponsorship.id, sponsor_id: sponsorship.sponsor_id, sponsorship: {end_date: Date.current}
+
+    expect(sponsorship.active).to be false
```

Comment:
I think the `#destroy` test below captures the essence of a controller spec better. The sponsorship's inactivation is performed by a service object, not the controller, which makes it a good candidate for stubbing just as `.destroy`. The controller only sends relevant instructions and decides what to do once it gets the response - whether the receivers were actually able to carry out those instructions is not its concern.

On the other hand, testing the unhappy paths of inactivation and destruction would be appropriate, as would testing that correct flash messages get set.